spock-core/src/main/java/org/spockframework/mock/runtime/ProxyBasedMockFactory.java

```diff
@@ -74,9 +90,47 @@ private Object createDynamicProxyMock(Class<?> mockType, List<Class<?>> addition
   }
 
   // inner class to defer class loading
-  private static class CglibMockFactory {
+  private static class ByteBuddyMockFactory {
+
+    private static final ByteBuddyClassCache CACHE = new ByteBuddyClassCache(true);
+
     static Object createMock(Class<?> type, List<Class<?>> additionalInterfaces, @Nullable List<Object> constructorArgs,
         IProxyBasedMockInterceptor interceptor, ClassLoader classLoader, boolean useObjenesis) {
+      Class<?> enhancedType = CACHE.find(classLoader, type, additionalInterfaces);
+
+      if (enhancedType == null) {
+        synchronized (type) {
```

Comment:
Yeah your are right, both approaches have their drawbacks. Could you document the rationale for this decision in a comment.